### DISCUSSION TITLE
Using DB_SOCKET

### DISCUSSION THREAD

**Discussion by trallen** [firefly-iii/firefly-iii#5647] [Firefly III]
I see that config/database.php lists the variable DB_SOCKET for MySQL databases; however, the unix_socket variable that it sets doesn't appear to be used anywhere else (notably, when creating the DSN in CreateDatabase.php: only DB_HOST and DB_PORT are used).

Using Docker, wait-for-it.sh also does not seem to respect sockets.

Is using a unix socket to access MySQL a supported configuration?

**Comment by JC5** [comment]
`wait-for-it.sh` does not work for sockets no. But I expect it's not a problem, the use case for `wait-for-it.sh` is primarily for Docker combined with its own database.

For the DSN in CreateDatabase.php I'll do some research to make sure it's also supported.

Otherwise it's supported but you're the only one I know that uses it 😁
